drivers/staging/vc04_services/interface/vchiq_arm/vchiq_core.c

```diff
@@ -498,6 +498,7 @@ vchiq_set_conn_state(struct vchiq_state *state, enum vchiq_connstate newstate)
 	vchiq_platform_conn_state_changed(state, oldstate, newstate);
 }
 
+/* FIXME */
```

Comment:
Remote events provide a way of presenting several virtual doorbells to a peer (ARM host to VPU) using only one physical doorbell. They can be thought of as a way for the peer to signal a semaphore, in this case implemented as a workqueue.

Remote events remain signalled until acknowledged by the receiver, and they are non-counting. They are designed in such a way as to minimise the number of interrupts and avoid unnecessary waiting.

A remote_event is as small data structures that live in shared memory. It comprises two booleans - armed and fired:
* The sender sets fired when they signal the receiver.
* If fired is set, the receiver has been signalled and need not wait.
* The receiver sets the armed field before they begin to wait.
* If armed is set, the receiver is waiting and wishes to be woken by interrupt.

remote_event_create initialises a single remote_event, and the associated wait_queue.